vonk/deployment/performance.rst

```diff
@@ -0,0 +1,161 @@
+
+Search Parameters
+~~~~~~~~~~~~~~~~~
+
+When a resource is sent to Vonk for storage, Vonk indexes the resource for all the search parameters that are applicable to the resource. The more search parameters are known to Vonk, the more resources will be used for indexing. Both time for the indexing processing and storage for storing the values for each search parameter. This also increases the size of the index tables and indexes, and therefore querying times. So if you know you will use only part of the predefined search parameters you can choose to delete the others from the Administration API. See :ref:`conformance` and :ref:`supportedmodel`.
+
+Pipeline
+~~~~~~~~
+
+Vonk is made up of a pipeline of components. You can leave out any component that you don't need. So if you don't need conditional processing (create, update, delete), just exclude them from the pipeline. Excluded components are not loaded and thus never executed. See :ref:`settings_pipeline`.
+
+Platform
+^^^^^^^^
+
+Vonk can run on Windows, Linux and MacOS. Directly or in a Docker container. On real hardware, virtual machine, app service or Kubernetes cluster. And then you can choose the dimensions of the platform, scaling up (bigger 'hard'ware) or scaling out (more (virtual) machines) as you see fit. Each of these choices will influence performance.
+
+Besides the way Vonk is deployed, the way the database is deployed is an important factor. Vonk needs a very low latency connection between the Vonk process(es) and the database server. If you have configured :ref:`configure_log_insights`, the calls to the database are recorded as separate dependencies so you can check whether this may be a bottleneck.
+
+Vonk is optimized for multithreaded processing. This means that it will benefit from extra processing cores, either in the same machine (multi core processor) or by adding additional machines (and thus processors). 
+
+Vonk is fully stateless, so no data about connections is saved between requests. First of all this helps in scaling out, since you don't need thread affinity. On top of that this limits the memory requirements of Vonk.
+
+Usage patterns
+^^^^^^^^^^^^^^
+
+How will Vonk be used in your environment? 
+
+#. Mostly for querying or rather for creating and updating resources?
+   Altering resources requires more processing than reading them. Also see the comment on indexing and search parameters above.
+#. How is the distribution of values in the resources that you query on?
+   E.g. if you use only a few types of resources, query them just by tag and the resources have only about 5 different tags, calculating the number of results will take a lot of time. Using more finegrained distributed values to query on solves this.
```

Comment:
I don't understand this, could you revise?

Reply:
I may just remove the last item - it is too detailed, and hopefully largely solved by revised indexes.